app/models/provider.rb

```diff
@@ -40,6 +40,9 @@
 #  index_provider_on_recruitment_cycle_id_and_provider_code  (recruitment_cycle_id,provider_code) UNIQUE
 #
 
+# Unpleasant hack to stop autoload error on CI
+require_relative "../services/providers/generate_course_code_service"
```

Comment:
Really not sure about this. How is it working now?

Reply:
We don't know why it's broken on CI without this, I went over it with Misha on Friday and didn't want to spend more time on it, we've got a ticket to investigate it: https://trello.com/c/9eWQ9rg3/2686-figure-out-why-this-commit-is-failing-on-azure-but-not-locally

Reply:
Ya, we started off with replaying the steps that the Azure pipeline followed to recreate this error, but ran out of steam and figured it would be better to kick that can further down the road as tech debt. I'm ok with this hack until we get time to further investigate it.